root_draft_metadata.tf

```diff
@@ -46,9 +47,158 @@ module "draft_metadata_api_gateway" {
   }]
 }
 
+resource "aws_iam_role" "draft_metadata_api_gateway_execution_role" {
+  name               = "TDRMetadataChecksAPIGatewayExecutionRole${title(local.environment)}"
+  assume_role_policy = templatefile("./templates/iam_policy/assume_role_policy.json.tpl", { service = "apigateway.amazonaws.com" })
+
+  inline_policy {
+    name   = "TDRMetadataChecksAPIGatewayStepFunctionExecutionPolicy${title(local.environment)}"
+    policy = templatefile("./templates/iam_policy/api_gateway_state_machine_policy.json.tpl", { account_id = data.aws_caller_identity.current.account_id, state_machine_arn = module.draft_metadata_checks.step_function_arn })
+  }
+}
+
 module "draft_metadata_bucket" {
   source      = "./da-terraform-modules/s3"
   bucket_name = local.draft_metadata_s3_bucket_name
   common_tags = local.common_tags
   kms_key_arn = module.s3_internal_kms_key.kms_key_arn
 }
+
+data "aws_ssm_parameter" "backend_checks_keycloak_secret" {
+  name            = local.keycloak_backend_checks_secret_name
+  with_decryption = true
+}
+
+resource "aws_cloudwatch_event_connection" "consignment_api_connection" {
+  name               = "TDRConsignmentAPIConnection${title(local.environment)}"
+  authorization_type = "OAUTH_CLIENT_CREDENTIALS"
+
+  auth_parameters {
+    oauth {
+      client_parameters {
+        client_id     = local.keycloak_backend-checks_client_id
+        client_secret = data.aws_ssm_parameter.backend_checks_keycloak_secret.value
+      }
+
+      authorization_endpoint = "${local.keycloak_auth_url}/realms/tdr/protocol/openid-connect/token"
+      http_method            = "POST"
+
+      oauth_http_parameters {
+        body {
+          key             = "grant_type"
+          value           = "client_credentials"
+          is_value_secret = false
+        }
+      }
+    }
+  }
+}
+
+resource "aws_iam_policy" "draft_metadata_checks_policy" {
+  name        = "TDRMetadataChecksPolicy${title(local.environment)}"
+  description = "Policy to allow necessary lambda executions from step function"
+
+  policy = templatefile("./templates/iam_policy/invoke_lambda_policy.json.tpl", {
+    resources = jsonencode([
+      module.yara_av_v2.lambda_arn,
+      module.draft_metadata_validator_lambda.lambda_arn
+    ])
+  })
+}
+
+resource "aws_iam_policy" "api_invoke_policy" {
+  name = "TDRAPIInvokePolicy${title(local.environment)}"
+
+  policy = templatefile("./templates/iam_policy/third_party_api_invocation_template.json.tpl", {
+    region            = local.region
+    account_number    = var.tdr_account_number
+    connection_arn    = aws_cloudwatch_event_connection.consignment_api_connection.arn
+    api_url           = module.consignment_api.api_url
+    step_function_arn = module.draft_metadata_checks.step_function_arn
+  })
+}
+
+module "draft_metadata_checks" {
+  source             = "./da-terraform-modules/sfn"
+  step_function_name = "TDRMetadataChecks${title(local.environment)}"
+  step_function_definition = jsonencode(
```

Comment:
The step function definition should go in a `.json.tpl` file. See here: https://github.com/nationalarchives/tdr-terraform-environments/tree/master/templates/step_function